### PR TITLE
Add profile screen and authorization screen

### PR DESCRIPTION
**Pull request by DmitryLorents**
https://github.com/DmitryLorents/Recepies/assets/119749906/b2994ae1-440c-4760-a570-800db2ee7007

https://github.com/DmitryLorents/Recepies/assets/119749906/480cb5a4-fb9f-493e-ae5b-3de8db2acee6
IMG_1023
IMG_1024
IMG_1025
IMG_1026
IMG_1027

### CODE REVIEW

File: Recepies/Modules/Profile/Presenter/PresenterProfile.swift [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    /// Инициализация профиля
+    init(view: ProfileViewProtocol, coordinator: BaseModuleCoordinator)
+    /// Массив опций
+    var options: [OptionsProtocol] { get set }
+    /// Информация о пользователе
+    var profileUser: ProfileUserProtocol { get set }
+    /// Загрузка алерта со сменой имени
+    func setupAlert()
+    /// Изменение имени в профиле
+    /// - Parameter :  name - Имя пользователя
```
Comment (Evgeny-Zverik) [inline Recepies/Modules/Profile/Presenter/PresenterProfile.swift]:
```suggestion

```
параметры не надо

File: Recepies/Modules/Profile/View/MainTableViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+    // MARK: - Visual Components
+
+    private let avatarImageView: UIImageView = {
+        let imageView = UIImageView()
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        return imageView
+    }()
+
+    private let fullName: UILabel = {
```
Comment (Evgeny-Zverik) [inline Recepies/Modules/Profile/View/MainTableViewCell.swift]:
```suggestion
    private let fullNameLabel: UILabel = {
```

File: Recepies/Modules/Profile/View/ProfileView.swift [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+    }
+
+    // MARK: - Public Properties
+
+    var profilePresenter: ProfilePresenter?
+
+    // MARK: - Private Properties
+
+    private var tableView = UITableView()
+    private let content: [CellTypes] = [.profile, .options]
```
Comment (Evgeny-Zverik) [inline Recepies/Modules/Profile/View/ProfileView.swift]:
```suggestion
    private let contentTypes: [CellTypes] = [.profile, .options]
```

**Review by Evgeny-Zverik — APPROVED** [review]

File: Recepies/App/Builder.swift [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        let profileView = ProfileView()
+        let profilePresenter = ProfilePresenter(view: profileView, coordinator: coordinator)
+        profileView.profilePresenter = profilePresenter
+        profileView.tabBarItem = UITabBarItem(
+            title: Constants.profileViewTitle,
+            image: .prof,
+            selectedImage: .profileSet
+        )
+        return profileView
+    }
```
Comment (DmitryLorents) [inline Recepies/App/Builder.swift:60]: ok

**Review by DmitryLorents — COMMENTED** [review]
Ok

File: Recepies/App/Builder.swift
Comment (DmitryLorents) [inline Recepies/App/Builder.swift:1]: ok

**Review by DmitryLorents — COMMENTED** [review]
1

File: Recepies/App/SceneDelegate.swift
Comment (DmitryLorents) [inline Recepies/App/SceneDelegate.swift:1]: 1

**Review by DmitryLorents — COMMENTED** [review]
1

**Pull request merged**